File: 백준/Bronze/1978. 소수 찾기/소수 찾기.cc
#include <stdio.h>
int main() {
	int n,m,ck=0,count=0;
	scanf("%d", &n);
	for (int i = 1; i <= n; i++) {
		scanf("%d", &m);
		if (m != 1) {
			for (int j = 2; j < m; j++) {
				if (m % j == 0)
					count++;
			}
			if (count == 0)
				ck++;
			count = 0;
		}
	}
	printf("%d", ck);
}

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.java

import java.util.Arrays;
import java.util.Scanner;

public class Main {
    static boolean[] sieve(int max) {
        boolean[] isPrime = new boolean[max + 1];
        Arrays.fill(isPrime,true);
        isPrime[0] =isPrime[1]=false;

        for (int p = 2; p * p <= max; p++) {
            if (isPrime[p]) {
                for (int m = p * p; m <= max; m += p) {
                    isPrime[m] = false;
                }
            }
        }
        return isPrime;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();
        int[] arr = new int[N];
        int max = 0;

        for (int i = 0; i < N; i++) {
            arr[i] = sc.nextInt();
            if (arr[i] > max) max = arr[i];
        }

        boolean[] isPrime = sieve(max);

        int count = 0;
        for (int x : arr) {
            if (isPrime[x]) count++;
        }
        System.out.println(count);
    }
}